client/mm/libxc/mexc.go

```diff
@@ -0,0 +1,3248 @@
+				rate: calc.MessageRateAlt(price, b.baseConversionFactor, b.quoteConversionFactor),
+				qty:  uint64(qty * float64(b.baseConversionFactor)),
+			})
+		}
+
+		return convertedUpdates, nil
+	}
+
+	bids, err = convert(mexcBids)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	asks, err = convert(mexcAsks)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return bids, asks, nil
+}
+
+// convertBookUpdate converts a BookUpdate with float64 arrays to obEntry slices.
+func (b *mexcOrderBook) convertBookUpdate(update *mxctypes.BookUpdate) (bids, asks []*obEntry, err error) {
+	convert := func(updates [][2]float64) ([]*obEntry, error) {
+		convertedUpdates := make([]*obEntry, 0, len(updates))
+
+		for _, update := range updates {
+			price := update[0]
+			qty := update[1]
+
+			convertedUpdates = append(convertedUpdates, &obEntry{
+				rate: calc.MessageRateAlt(price, b.baseConversionFactor, b.quoteConversionFactor),
+				qty:  uint64(qty * float64(b.baseConversionFactor)),
+			})
+		}
+
+		return convertedUpdates, nil
+	}
+
+	bids, err = convert(update.Bids)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	asks, err = convert(update.Asks)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return bids, asks, nil
+}
+
+// Connect implements the dex.Connector interface.
+// Synchronizes orderbook: buffer WS updates, fetch REST snapshot, apply buffered updates.
+func (b *mexcOrderBook) Connect(ctx context.Context) (*sync.WaitGroup, error) {
+	const updateIDUnsynced = math.MaxUint64
+
+	// Synchronization variables
+	var syncMtx sync.Mutex
+	var syncCache []*mxctypes.BookUpdate
+	syncChan := make(chan struct{})
+	b.syncChan = syncChan
+	var updateID uint64 = updateIDUnsynced
+	acceptedUpdate := false
+
+	resyncChan := make(chan struct{}, 1)
+
+	desync := func(resync bool) {
+		// Clear the sync cache, set the special ID, trigger a book refresh.
+		syncMtx.Lock()
+		defer syncMtx.Unlock()
+		syncCache = make([]*mxctypes.BookUpdate, 0)
+		acceptedUpdate = false
+		if updateID != updateIDUnsynced {
+			b.synced.Store(false)
+			updateID = updateIDUnsynced
+			// Close old syncChan and create new one for next sync
+			if syncChan != nil {
+				close(syncChan)
+			}
+			syncChan = make(chan struct{})
+			b.syncChan = syncChan
+			if resync {
+				select {
+				case resyncChan <- struct{}{}:
+				default:
+				}
+			}
+		}
+	}
+
+	acceptUpdate := func(update *mxctypes.BookUpdate) bool {
+		if updateID == updateIDUnsynced {
+			// Book is still syncing. Add it to the sync cache.
+			syncCache = append(syncCache, update)
+			return true
+		}
+
+		// MEXC protobuf doesn't provide sequence IDs, accept all updates after sync.
+		if !acceptedUpdate {
+			acceptedUpdate = true
+		}
+
+		bids, asks, err := b.convertBookUpdate(update)
+		if err != nil {
+			b.log.Errorf("Error parsing MEXC book update: %v", err)
+			return false
+		}
+		// MEXC sends full snapshots; clear stale data before applying.
+		b.book.clear()
+		b.book.update(bids, asks)
+		return true
```

Comment:
If I understand this correctly, you don't need a syncCache? You only need to save the last book update while desynced?